packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -1326,19 +1319,9 @@ export class UmbrellaStore<M extends BaseMetadata> {
     query: ThreadsQuery<M> | undefined
   ) {
     const threadsFetcher = async (cursor?: string) => {
-      if (this._client === undefined) {
-        // TODO: Think about other ways to structure this. Throwing a StopRetrying only
-        // makes sense only if we can easily know if the fetcher is going to be wrapped inside the autoRetry function
-        throw new StopRetrying(
-          "Client is required in order to load threads for the room"
-        );
-      }
```

Comment:
I made `this._client` mandatory, so we don't have to worry about these edge cases anymore. This ugliness is now shifted to the unit tests. Turns out there were only two direct creations of `new UmbrellaStore()` left in the unit tests, so it really wasn't a big deal anymore.